### PR TITLE
Add Flow Control to PubSub consumer

### PR DESCRIPTION
**Pull request by oguzhanunlu**
- [x] Reference value is to be updated after benchmarking

### CODE REVIEW

File: modules/pubsub/src/main/resources/application.conf [hunk 1]
```diff
@@ -3,6 +3,8 @@
     "type": "PubSub"
     "parallelPullCount": 1
     "maxQueueSize": 3000
```
Comment (benjben) [inline modules/pubsub/src/main/resources/application.conf:5]: I would set that much higher like 10k or maybe even 100k, given that we have the bytes limit

File: modules/pubsub/src/main/scala/com/snowplowanalytics/snowplow/enrich/pubsub/Source.scala [hunk 2]
```diff
@@ -44,12 +47,24 @@ object Source {
...
+        .setMaxOutstandingRequestBytes(input.maxRequestBytes)
+        .setLimitExceededBehavior(LimitExceededBehavior.Block)
+        .build()
+
     val pubSubConfig =
       PubsubGoogleConsumerConfig(
         onFailedTerminate = onFailedTerminate,
         parallelPullCount = input.parallelPullCount,
-        maxQueueSize = input.maxQueueSize
+        maxQueueSize = input.maxQueueSize,
```
Comment (benjben) [inline modules/pubsub/src/main/scala/com/snowplowanalytics/snowplow/enrich/pubsub/Source.scala:63]: Do we still need this line?

**Review by benjben — APPROVED** [review]
:+1:

File: modules/pubsub/src/main/scala/com/snowplowanalytics/snowplow/enrich/pubsub/Source.scala (hunk 2 above)
Comment (oguzhanunlu) [reply modules/pubsub/src/main/scala/com/snowplowanalytics/snowplow/enrich/pubsub/Source.scala:63]: yes, to set size of fs2-pubsub's internal queue

File: modules/pubsub/src/main/resources/application.conf (hunk 1 above)
Comment (oguzhanunlu) [reply modules/pubsub/src/main/resources/application.conf:5]: going for 10K after the test against 1KB events

File: config/config.pubsub.extended.hocon [hunk 3]
```diff
@@ -17,6 +17,16 @@
     # enough that it does not cause the sink to block whilst it is waiting for a batch to be
     # completed.
     "maxQueueSize": 3000
```
[the remark this answers is not in the document]
Comment (oguzhanunlu) [reply config/config.pubsub.extended.hocon:19]: right, thanks Ben!

File: modules/pubsub/src/main/scala/com/snowplowanalytics/snowplow/enrich/pubsub/Source.scala (hunk 2 above)
Comment (benjben) [reply modules/pubsub/src/main/scala/com/snowplowanalytics/snowplow/enrich/pubsub/Source.scala:63]: Ah yes I now get the difference, thanks!

**Review by istreeter — APPROVED** [review]

File: modules/pubsub/src/main/resources/application.conf (hunk 1 above)
Comment (oguzhanunlu) [reply modules/pubsub/src/main/resources/application.conf:5]: decided to stay with 3K after observing memory usage

**Pull request merged**